### PR TITLE
Add initial Abstract Syntax Tree

### PR DESCRIPTION
**Pull request by tomekb234**

### CODE REVIEW

File: app/src/main/kotlin/compiler/ast/StatementAst.kt [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    data class Loop(
+        val condition: ExpressionAst,
+        val action: BlockAst
+    ) : StatementAst()
+
+    object LoopBreak : StatementAst()
+
+    object LoopContinuation : StatementAst()
+
+    data class FunctionReturn(val value: ExpressionAst?) : StatementAst()
```
Comment (ggawryal) [inline app/src/main/kotlin/compiler/ast/StatementAst.kt:33]: Shouldn't we always return some value (for `Unit` functions `zakończ` will be translated to `zwróć Unit`)?

File: app/src/main/kotlin/compiler/ast/ExpressionAst.kt [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    data class UnaryOperation(
+        val kind: Kind,
+        val operand: ExpressionAst
+    ) : ExpressionAst() {
+        enum class Kind {
+            NOT,
+            NEG,
+            BIT_NOT,
+        }
```
Comment (ggawryal) [inline app/src/main/kotlin/compiler/ast/ExpressionAst.kt:28]: Just to confirm: we don't want to have an unary `plus`, right?

File: app/src/main/kotlin/compiler/ast/StatementAst.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
+package compiler.ast
+
+typealias BlockAst = List<StatementAst>
+
+sealed class StatementAst {
```
Comment (ggawryal) [inline app/src/main/kotlin/compiler/ast/StatementAst.kt:5]: Sometimens `Comments` are also included in AST, do we want to add them?

**Review by ggawryal — APPROVED** [review]
Dzięki, wygląda dobrze. Zostawiłem kilka pytań bardziej odnośnie języka niż kodu.

File: app/src/main/kotlin/compiler/ast/StatementAst.kt (hunk 1 above)
Comment (tomekb234) [reply app/src/main/kotlin/compiler/ast/StatementAst.kt:33]: I think this translation should be performed by the type checker, and not by parser. The AST should reflect the code as seen by the programmer. For the same reason the field `returnType` in `FunctionAst` is optional.

File: app/src/main/kotlin/compiler/ast/ExpressionAst.kt (hunk 2 above)
Comment (tomekb234) [reply app/src/main/kotlin/compiler/ast/ExpressionAst.kt:28]: We can add it, but I suppose the language won't become unusable without it.

File: app/src/main/kotlin/compiler/ast/StatementAst.kt (hunk 3 above)
Comment (tomekb234) [reply app/src/main/kotlin/compiler/ast/StatementAst.kt:5]: The compiler probably won't make any use of them, so I don't see a point.

File: app/src/main/kotlin/compiler/ast/ExpressionAst.kt (hunk 2 above)
Comment (tomekb234) [reply app/src/main/kotlin/compiler/ast/ExpressionAst.kt:29]: Added, and made the operation names more readable.

**Pull request merged**